tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -110,6 +110,23 @@ var _ = Describe("odo devfile push command tests", func() {
 			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
 		})
 
+		When("Devfile 2.1.0 is used", func() {
+			It("should work with devfile variables", func() {
+				helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, cmpName)
+
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-variables.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+
+				output := helper.CmdShouldPass("odo", "push")
+				Expect(output).To(ContainSubstring("Changes successfully pushed to component"))
+				routeURL := helper.DetermineRouteURL(commonVar.Context)
```

Comment:
I don't think it will work as the url host is not present

Reply:
hmm, good point. This will work only on OpenShift :-( 
I will look into this.